src/main/java/ui/components/PanelMenuBarWithRename.java

```diff
@@ -0,0 +1,105 @@
+package ui.components;
+import backend.interfaces.IModel;
+import javafx.scene.control.Label;
+import ui.issuepanel.FilterPanel;
+import javafx.application.Platform;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.HBox;
+
+/**
+ * Created by manmeetsinghsaluja on 23/11/15.
+ */
+public class PanelMenuBarWithRename extends HBox {
+
+    private HBox menuBarUndoButton;
+    private HBox menuBarConfirmButton;
+    private HBox menuBarCloseButton;
+    private TextField renameTextField;
+    private HBox menuBarNameArea;
+    private HBox nameBox;
+    private Label renameButton;
+    private FilterPanel panel;
+    private IModel model;
+    private int panelIndex;
+
+    public PanelMenuBarWithRename(FilterPanel panel, IModel model, int panelIndex){
+        this.panel = panel;
+        this.model = model;
+        this.panelIndex = panelIndex;
+        this.menuBarUndoButton = panel.menuBarUndoButton;
+        this.menuBarCloseButton = panel.menuBarCloseButton;
+        this.menuBarConfirmButton = panel.menuBarConfirmButton;
+        this.menuBarNameArea = panel.menuBarNameArea;
+        this.nameBox = panel.nameBox;
+        this.renameButton = panel.renameButton;
+    }
+
+    public void initializeRenameTextFieldAndEvents() {
+        renameTextField = new TextField();
```

Comment:
`renameTextField` sounds like a method rather than a variable. Should use a noun. e.g. `renamableTextField`